flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/DefaultResourceCleaner.java

```diff
@@ -146,14 +146,15 @@ private Builder(
          */
         public Builder<T> withRegularCleanup(String label, T regularCleanup) {
             assertUniqueLabel(label);
-            this.regularCleanup.put(label, regularCleanup);
+            this.regularCleanup.add(new CleanupWithLabel<>(regularCleanup, label));
             return this;
         }
 
         private void assertUniqueLabel(String label) {
             Preconditions.checkArgument(
-                    !this.prioritizedCleanup.containsKey(label)
-                            && !this.regularCleanup.containsKey(label),
+                    Stream.concat(prioritizedCleanup.stream(), regularCleanup.stream())
+                            .map(CleanupWithLabel::getLabel)
+                            .noneMatch(label::equals),
```

Comment:
Why should they still be required to be unique?

Reply:
yikes, I didn't go through it properly. I added it because of the Maps I previously used. I agree that it's not necessary with the Pojo approach 👍